test/lsp/test_read_lsp_diagnostics.vader

```diff
@@ -109,6 +109,26 @@ Execute(ale#lsp#response#ReadDiagnostics() should include sources in detail):
   \   }
   \ ]}})
 
+Execute(ale#lsp#response#ReadDiagnostics() should keep detail with line breaks but replace with spaces in text):
+  AssertEqual [
+  \   {
+  \     'type': 'E',
+  \     'text': 'cannot borrow `cap` as mutable more than once at a time  mutable borrow starts here in previous iteration of loop',
+  \     'detail': '[rustc] cannot borrow `cap` as mutable more than once at a time\n\nmutable borrow starts here in previous iteration of loop',
+  \     'lnum': 10,
+  \     'col': 15,
+  \     'end_lnum': 12,
+  \     'end_col': 22,
+  \   }
+  \ ],
+  \ ale#lsp#response#ReadDiagnostics({'params': {'uri': 'filename.ts', 'diagnostics': [
+  \   {
+  \     'range': Range(9, 14, 11, 22),
+  \     'message': 'cannot borrow `cap` as mutable more than once at a time\n\nmutable borrow starts here in previous iteration of loop',
```

Comment:
```suggestion
  \     'message': "cannot borrow `cap` as mutable more than once at a time\n\nmutable borrow starts here in previous iteration of loop",
```

Here's the problem with your test. Escapes like `\n` are only applied for double quoted strings.